GherkinSyncTool.Synchronizers.AzureDevOps/Client/AzureDevopsClient.cs

```diff
@@ -205,5 +195,60 @@ private WorkItemTrackingHttpClient GetWorkItemTrackingHttpClient()
 
             return workItemTrackingHttpClient;
         }
+
+        private IEnumerable<WorkItemReference> GetListOfWorkItemsWithQuery(string query)
+        {
+            var results = new List<WorkItemReference>();
+            var workItemTrackingHttpClient = GetWorkItemTrackingHttpClient();
+            var counter = 10000;
+            var moreResults = true;
+            
+            while (moreResults)
+            {
+                // wiql - Work Item Query Language
+                var wiql = new Wiql
+                {
+                    Query = $@"{query}
+                           AND [{WorkItemFields.Id}] >= {counter - 10000} 
+                           AND [{WorkItemFields.Id}] < {counter}"
+                };
+            
+                var currentResults = workItemTrackingHttpClient.QueryByWiqlAsync(wiql, _azureDevopsSettings.Project)
+                    .Result.WorkItems.ToList();
```

Comment:
Instead of calling `ToList()` on collection and calling its `Count` property, try to use `Any()` - this way you will know if collection has any items and you'll avoid enumeration through all of the elements

Reply:
I agree with `Any()`, but if I do not convert `IEnumerable `to `List`, won't it cause multiple enumeration? because it works with query data, which can change between calls, isn't using `ToList()` on the safer side, so I won't get "dirty data", which I might get between `GetEnumerator()` calls ?

Reply:
If this is the case then I suggest change method return type to IList

Reply:
Done